mobile/android/android-debug-bridge.ts

```diff
@@ -11,35 +11,34 @@ interface IComposeCommandResult {
 export class AndroidDebugBridge implements Mobile.IAndroidDebugBridge {
 	constructor(private identifier: string,
 		private $childProcess: IChildProcess,
-		private $errors: IErrors,	
-		private $logger: ILogger,	
+		private $errors: IErrors,
+		private $logger: ILogger,
 		private $staticConfig: Config.IStaticConfig) { }
-	
 
-	public executeCommand(...args: string[]): IFuture<any> {
+
+	public executeCommand(args: string[]): IFuture<any> {
 		return (() => {
 			let command = this.composeCommand(args).wait();
-			return this.$childProcess.spawnFromEvent(command.command, command.args, "close").wait().stdout;
+			return this.$childProcess.spawnFromEvent(command.command, command.args, "close", undefined, {throwError: false}).wait().stdout;
 		}).future<any>()();
 	}
-	
-	public executeShellCommand(...args: string[]): IFuture<any> {
+
+	public executeShellCommand(args: string[]): IFuture<any> {
 		return (() => {
 			args.unshift("shell");
 			let shellCommand = this.composeCommand(args).wait();
-			this.$logger.trace(`Shell command ${shellCommand}`);
-			return this.$childProcess.spawnFromEvent(shellCommand.command, shellCommand.args, "close").wait().stdout;
+			return this.$childProcess.spawnFromEvent(shellCommand.command, shellCommand.args, "close", undefined, {throwError: false}).wait().stdout;
 		}).future<any>()();
 	}
-	
+
 	public sendBroadcastToDevice(action: string, extras: IStringDictionary = {}): IFuture<number> {
 		return (() => {
-			let broadcastCommand = `am broadcast -a "${action}"`;
-			_.each(extras, (value,key) => broadcastCommand += ` -e "${key}" "${value}"`);
+			let broadcastCommand = ["am", "broadcast", "-a", `${action}`];
```

Comment:
is there any chance to have space in "action", as far as I see from the previous code, it was intended to have "", so consider using `"${action}"`